kernel/src/table_properties/deserialize.rs

```diff
@@ -0,0 +1,287 @@
+    }
+
+    fn next_value_seed<V>(&mut self, seed: V) -> Result<V::Value, Self::Error>
+    where
+        V: DeserializeSeed<'de>,
+    {
+        let value = self.value.take().unwrap(); // FIXME
+        let de = de::value::StrDeserializer::new(value);
+        seed.deserialize(de)
+    }
+}
+
+pub(crate) fn deserialize_option<'de, D, T>(deserializer: D) -> Result<Option<T>, D::Error>
+where
+    D: Deserializer<'de>,
+    T: Deserialize<'de>,
+{
+    T::deserialize(deserializer).map(Some)
+}
+
+pub(crate) fn deserialize_pos_int<'de, D>(deserializer: D) -> Result<Option<u64>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+    let n: u64 = s.parse().unwrap(); // FIXME
+    if n == 0 {
+        panic!("FIXME");
+        // return Err("something");
+    }
+    Ok(Some(n))
+}
+
+pub(crate) fn deserialize_bool<'de, D>(deserializer: D) -> Result<Option<bool>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+    match s.as_str() {
+        "true" => Ok(Some(true)),
+        "false" => Ok(Some(false)),
+        _ => Err(de::Error::unknown_variant(&s, &["true", "false"])),
+    }
+}
+
+pub(crate) fn deserialize_column_names<'de, D>(
+    deserializer: D,
+) -> Result<Option<Vec<ColumnName>>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: String = Deserialize::deserialize(deserializer)?;
+    Ok(Some(
+        s.split(',')
+            .map(|name: &str| ColumnName::new([name]))
+            .collect(),
+    ))
+}
+
+pub(crate) fn deserialize_interval<'de, D>(deserializer: D) -> Result<Option<Duration>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s = String::deserialize(deserializer)?;
+    parse_interval(&s).map(Some).map_err(de::Error::custom)
+}
+
+fn parse_interval(value: &str) -> Result<Duration, String> {
+    let not_an_interval = || format!("'{value}' is not an interval");
+
+    if !value.starts_with("interval ") {
+        return Err(not_an_interval());
+    }
+    let mut it = value.split_whitespace();
+    let _ = it.next(); // skip "interval"
+    let number = parse_int(it.next().ok_or_else(not_an_interval)?)?;
+    if number < 0 {
+        return Err(format!("interval '{value}' cannot be negative"));
+    }
+    let number = number as u64;
+
+    let duration = match it.next().ok_or_else(not_an_interval)? {
+        "nanosecond" | "nanoseconds" => Duration::from_nanos(number),
+        "microsecond" | "microseconds" => Duration::from_micros(number),
+        "millisecond" | "milliseconds" => Duration::from_millis(number),
+        "second" | "seconds" => Duration::from_secs(number),
+        "minute" | "minutes" => Duration::from_secs(number * SECONDS_PER_MINUTE),
+        "hour" | "hours" => Duration::from_secs(number * SECONDS_PER_HOUR),
+        "day" | "days" => Duration::from_secs(number * SECONDS_PER_DAY),
+        "week" | "weeks" => Duration::from_secs(number * SECONDS_PER_WEEK),
```

Comment:
Is this list coming from somewhere specific that we need to stay in sync with?
(I wonder e.g. if we need to support e.g. `nanos` and `micros`)

Probably worth documenting the connection.

Reply:
this came from `delta-rs` (recycled from Robert's old PR) but I've documented that in the code and pointed out the differences between this and delta-spark's behavior. also created a follow-up to improve this but i think we can leave this here for now and improve in follow-up? #507